apps/public-reference/cypress/integration/pages/application/contact/name.ts

```diff
@@ -0,0 +1,50 @@
+describe("Form contact/name", function () {
+  it("Renders the /contact/name form", function () {
+    cy.visit("/applications/contact/name")
+    cy.get("form").should("be.visible")
+    cy.location("pathname").should("include", "/applications/contact/name")
+  })
+
+  it("Should display initial form errors", function () {
+    // try to trigger form
```

Comment:
These tests are tightly coupled to the order in which they're defined inside this file - this isn't ideal, since it means that a test will fail if you try to run it individually, or if someone unwittingly modifies the order down the road (e.g. removes this first test).

A better way to structure this would be to make sure that each test ensures a clean slate - either start it with a `cy.visit` call, or potentially add a [`beforeEach`](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Hooks) hook to this `describe` block.

Reply:
If the `onBeforeLoad` callback for `cy.visit` could be used to set an initial state of the application data via session storage, maybe that would work?

Reply:
Strange, I posted a comment and GitHub showed it twice, then I deleted one and both disappeared. Anyway…I was saying I think we could maybe leverage the `onBeforeLoad` callback for `cy.visit` so that session storage application data could be set for the context.

Reply:
@bencpeters **beforeEach** sounds good to me, good idea!